### ISSUE TITLE
No gradient following calibration tutorial

### ISSUE THREAD

**Issue by WinstonChenn** [AgentTorch/AgentTorch#55]
I'm trying to learn how to use this library. However, when following the [tutorial for calibrating a model with internal parameter](https://agenttorch.github.io/AgentTorch/tutorials/calibrating-a-model/#method-1-internal-parameter-optimization), I used the following code, exactly copied from the tutorial:
```
import warnings
warnings.simplefilter("ignore")

import torch
import torch.nn as nn
from AgentTorch.agent_torch.models import covid
from AgentTorch.agent_torch.populations import sample
from AgentTorch.agent_torch.core.executor import Executor
from AgentTorch.agent_torch.core.dataloader import LoadPopulation

# Initialize simulation
sim = Executor(covid, pop_loader=LoadPopulation(sample))
runner = sim.runner
runner.init()

def execute(runner, n_steps=5):
    """Execute simulation and compute loss"""
    runner.step(n_steps)
    labels = runner.state_trajectory[-1][-1]['environment']['daily_infected']
    return labels.sum()

def optimize_internal_params():
    # Execute simulation and compute gradients
    loss = execute(runner)
    loss.backward()

    # Get gradients of learnable parameters
    learn_params_grad = [(name, param, param.grad) 
                        for (name, param) in runner.named_parameters()]
    return learn_params_grad

# Example usage
gradients = optimize_internal_params()
print("Internal parameter gradients:", gradients)
```

And I got following outputs:
```
Registry:  Registry()
Internal parameter gradients: [('initializer.transition_function.0.new_transmission.learnable_args.R2', Parameter containing:
tensor([[4.7500],
        [4.7500],
        [4.7500]], requires_grad=True), None), ('initializer.transition_function.1.seirm_progression.learnable_args.M', Parameter containing:
tensor([0.1200], requires_grad=True), None)]
```

This indicates that no gradient is being passed back to the parameters. Is there something wrong with the current tutorial example?

**Comment by ayushchopra96** [comment]
Hi @WinstonChenn apologies for delayed response.

The sim parameters are being passed as `torch.tensor` instead of `nn.Parameter` in the default code. Hence, you're not seeing the gradient. This is required when calibrating using mode 2.

You need to set the `calibration: true` variable in the config.yaml to the false - before you call runner.init() [1] - to see the gradient on the simulation parameter.

My apologies for the missing detail in the tutorial.

[1]: https://github.com/AgentTorch/AgentTorch/blob/ayan-initializer/agent_torch/models/covid/yamls/config.yaml#L23C3-L23C20
